Add android target recipes and extensive model tests using ios and android recipes

abhinaykukkadapu opened the pull request:
Summary:
Changes in this diff:
1. Adds android target recipes using QNN backend.
2. Add extensive model lowering and accuracy checks using torchvision models.

```
from executorch.export import export
qnn_android_recipe = get_android_recipe()

# lower a model to pte
session = export(model, recipe, example_inputs)

out = session.run_method("forward", *example_inputs)
```

Differential Revision: D82284987

cccclai commented on backends/qualcomm/_passes/TARGETS:
```diff
@@ -15,5 +15,6 @@ runtime.python_library(
         "//executorch/backends/transforms:decompose_sdpa",
         "//executorch/exir/backend:backend_details",
         "//executorch/exir/backend:compile_spec_schema",
+        "//executorch/backends/qualcomm/quantizer:quantizer",
```
oh.....I see yeah that makes sense

abhinaykukkadapu committed:
Add android target recipes and extensive model tests using ios and android recipes (#14290)

Summary:

Changes in this diff:
1. Adds android target recipes using QNN backend.
2. Add extensive model lowering and accuracy checks using torchvision models.

```
from executorch.export import export
qnn_android_recipe = get_android_recipe()

# lower a model to pte
session = export(model, recipe, example_inputs)

out = session.run_method("forward", *example_inputs)
```

Differential Revision: D82284987
